mongo/mongo.go

```diff
@@ -96,3 +97,33 @@ func (m *Mongo) UpdateRecipe(id string, update bson.M) (err error) {
 	}
 	return err
 }
+
+//UpdateRecipe prepares updates in bson.M and then updates existing recipe document
+func (m *Mongo) UpdateRecipe(recipeID string, updates recipe.Response) (err error) {
```

Comment:
I dont see any difference between UpdateRecipe and AddInstance, other than one being a pointer type. Would be cleaner if we just used one. If there are places where we're calling UpdateRecipe that are not just adding an instance then I'd be happy for us to keep UpdateRecipe and bin AddInstance.

Reply:
The UpdateRecipe function is essentially updating the entire recipe - which is what the AddInstance is doing as well, so we do not require AddInstance and it can be substituted with the UpdateRecipe instead

Done in commit "remove AddInstance mongo function and edit its test functions accordingly"